mirrord-layer/src/config.rs

```diff
@@ -25,4 +25,11 @@ pub struct Config {
 
     #[envconfig(from = "MIRRORD_FILE_OPS", default = "false")]
     pub enabled_file_ops: bool,
+
+    #[envconfig(from = "MIRRORD_OVERRIDE_ENV_VARS", default = "false")]
+    pub enabled_override_env_vars: bool,
+
+    /// Overrides local env variables with remote ones, except for those specified here.
+    #[envconfig(from = "MIRRORD_OVERRIDE_FILTER_ENV_VARS", default = "PATH;HOME")]
```

Comment:
tbh I'm not sure if when using the env var we want them to explicitly add those, I'd imagine that even if the env/cli argument is used the user wouldn't expect to need to specify those as it was "magically" added before.
I'd add PATH;HOME regardless of the env var, and maybe later on add a flag to remove those explicitly (can't imagine someone wanting the remote PATH?)

Reply:
Yeah, that's why I started out doing it the other way around (user has to specify what they want). 

I'll add these in some sort of default ignore when the env vars are loaded, and if someone ever asks to have more control we can have some other config like `OVERRIDE_EVERYTHING`.